### PR TITLE
Add @irvingjs/loadable package

### PR DESCRIPTION
**Pull request by ostowe**
## Issue(s): Relates to or closes...
[**IRV-684**](https://alleyinteractive.atlassian.net/browse/IRV-684)

## Summary: This pull request will...
* Add an `@irvingjs/loadable` package to support [`@loadable/component`](https://loadable-components.com/)
* Migrate from `react-hot-loader` to `react-fast-refresh` to support HMR with loadable components (and that's where the ecosystem is going anyway)
* Add `ssrTags`, `waitForClientRender` configuration fields

## Tests: I know this code works because...
Tested in a client project and irving example theme

### CODE REVIEW

File: packages/core/server/utils/getTemplateVars.js [hunk 1]
```diff
@@ -59,49 +59,65 @@ const stringifyHeadConfig = (config, key) => {
...
  * @return {object}
  */
-export default function getTemplateVars(key, initialVars) {
+export default function getTemplateVars(key, initialVars, clientStats) {
   const templateVars = getConfigValues(key);
+
   // Separate arrays for head configuration and other template variables,
   // as they will be extracted and merged using different methods.
-  const varsConfigs = [];
+  // const varsConfigs = [];
```
Comment (jameswburke) [inline packages/core/server/utils/getTemplateVars.js:71]: Kill this?

```suggestion
```

File: stories/10-packages/Loadable.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<Meta title="Packages|Loadable" />
```
Comment (jameswburke) [inline stories/10-packages/Loadable.stories.mdx:1]: Can you update the table of contents story for the section too?

File: stories/10-packages/Loadable.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+# Loadable
+This package adds functionality to Irving core for using the [`@loadable/component`](https://loadable-components.com/) package.
```
Comment (jameswburke) [inline stories/10-packages/Loadable.stories.mdx:4]: Some additional info about _why_ this is cool would be good.

File: stories/10-packages/Loadable.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+## Using this package
+Once configured, you can use any feature of the `@loadable/component` package. See the [API Docs](https://loadable-components.com/docs/api-loadable-component/) for more information. The most basic way to use it, however, is to modify your component map to use loadable components instead of statically-imported ones. Example:
+
+```javascript
+// componentMap.js
+import loadable from '@loadable/component';
+
+const componentMap = {
+    'my-project/test': loadable(() => import('./components/test')),
```
Comment (jameswburke) [inline stories/10-packages/Loadable.stories.mdx:30]: Is there some way we could offer a version of the styled components package that does this automatically?

Immediate thoughts on possibilities,
* Another export from `styled-components` for `createThemedAndLoadableComponentMap`
* An irving.config value that flags it to be true `{ useLoadableStyledComponents: true, }`

**Review by jameswburke — APPROVED** [review]
A lot of the deep webpack is beyond me, but I noted the thoughts I had. Probably wise to get another set of eyes from someone better versed in webpack.

:bee:

File: stories/10-packages/Loadable.stories.mdx (hunk 4 above)
Comment (ostowe) [reply stories/10-packages/Loadable.stories.mdx:30]: Yeah I could see just making a separate component map for this.

**Review by tylerreckart — APPROVED** [review]
🧑‍🚀 Looks good!

**Pull request merged**